src/sbml/SBMLErrorLog.cpp

```diff
@@ -406,7 +406,17 @@ SBMLErrorLog::getErrorWithSeverity(unsigned int n, unsigned int severity) const
   {
     if (matcher(*it))
     {
-      if (count == n) return dynamic_cast<const SBMLError*>(*it);
+      if (count == n)
+      {
+        if (fromCAPI == 0)
+        {
+          return dynamic_cast<const SBMLError*>(*it);
+        } 
+        else
+        {
+          return static_cast<const SBMLError*>(*it);
+        }
+      }
```

Comment:
this looks scary, is this really needed? Is it safe? because those could always be XMLErrors and not SBMLErrors, so when accessing them from the C API we'd be giving them bad pointers. It would be better to just use the get error function returning an XMLError and have a function to figure out whether an XMLError is an SBMLError (which would be clear from the error code)

I'm sure I could devise a way to make that static cast do bad things if needed. So let's think about this once again. (sorry)